### PR TITLE
Feat(drivers): Support set cookie for Cloudreve (#4324)

### PR DESCRIPTION
**Pull request by ty114514**
Feat(drivers): Support set cookie for Cloudreve #4324
As with my Issues, some of the Cloudreve sites have reCaptcha enabled, and this PR will support users setting cookies directly for mounting

### CODE REVIEW

File: drivers/cloudreve/util.go [hunk 1]
```diff
@@ -49,14 +51,18 @@ func (d *Cloudreve) request(method string, path string, callback base.ReqCallbac
...
+				if err != nil {
+					return err
+				}
+				return d.request(method, path, callback, out)
 			}
-			return d.request(method, path, callback, out)
+			// 错误影响正常访问，下线该储存
+			d.GetStorage().SetStatus(fmt.Sprintf("%+v", err.Error()))
+			op.MustSaveDriverStorage(d)
+			return errors.New(r.Msg)
```
Comment (Xhofe) [inline drivers/cloudreve/util.go:64]: This line should be in the original position

**Review by Xhofe — CHANGES_REQUESTED** [review]

File: drivers/cloudreve/util.go (hunk 1 above)
Comment (ty114514) [reply drivers/cloudreve/util.go:64]: No need to do when cookie expired?

Comment (Xhofe) [reply drivers/cloudreve/util.go:64]: if r.Code != 0, we should return an error whether the cookie expired rather than continue

Comment (ty114514) [reply drivers/cloudreve/util.go]: Done. I also submitted to frontend repo

**Pull request merged**